pkg/apis/ctrlmesh/v1alpha1/faultinjection_types.go

```diff
@@ -55,41 +57,114 @@ type HTTPFaultInjectionDelay struct {
+	URL []string `json:"url"`
+	// Method specifies the http method of the request, like: PUT, POST, GET, DELETE.
+	Method []string `json:"method"`
 }
 
-type StringMatchType string
+type HTTPMatchRequest struct {
+	Name             string           `json:"name,omitempty"`
+	RelatedResources []*ResourceMatch `json:"relatedResources,omitempty"`
 
-const (
-	ExactMatchType  StringMatchType = "Exact"
-	PrefixMatchType StringMatchType = "Prefix"
-	RegexMatchType  StringMatchType = "Regex"
+	RestRules []*MultiRestRule `json:"restRules,omitempty"`
+}
+
+// HTTPFaultInjection can be used to specify one or more faults to inject
+// while forwarding HTTP requests to the destination specified in a route.
+type HTTPFaultInjection struct {
+	// Name is the name of the policy
+	Name string `json:"name,omitempty"`
+	// Delay requests before forwarding, emulating various failures such as
+	// network issues, overloaded upstream service, etc.
+	Delay *HTTPFaultInjectionDelay `json:"delay,omitempty"`
+	// Abort Http request attempts and return error codes back to downstream
+	// service, giving the impression that the upstream service is faulty.
+	Abort *HTTPFaultInjectionAbort `json:"abort,omitempty"`
+
+	// Match specifies a set of criterion to be met in order for the
+	// rule to be applied to the HTTP request.
+	Match *HTTPMatchRequest `json:"match,omitempty"`
+	// Effective time of fault injection
+	EffectiveTime *EffectiveTimeRange `json:"effectiveTime,omitempty"`
+}
+
+type FaultInjectionSpec struct {
+	// Selector is a label query over pods of this configuration.
+	Selector *metav1.LabelSelector `json:"selector,omitempty"`
+
+	Disabled bool `json:"disabled,omitempty"`
+
+	HTTPFaultInjections []*HTTPFaultInjection `json:"httpFault,omitempty"`
+}
+
+type EffectiveTimeRange struct {
+    // StartTime is the starting time of fault injection.
+    StartTime string `json:"startTime,omitempty"`
+
+    // EndTime is the ending time of fault injection.
+    EndTime string `json:"endTime,omitempty"`
+
+    // DaysOfWeek specifies on which days of the week the fault injection configuration is effective.
+    // 0 represents Sunday, 1 represents Monday, and so on.
+    DaysOfWeek []int `json:"daysOfWeek,omitempty"`
+
+    // DaysOfMonth specifies on which days of the month the fault injection configuration is effective.
+    // For example, 1 represents the first day of the month, and so on.
+    DaysOfMonth []int `json:"daysOfMonth,omitempty"`
+
+    // Months specifies in which months of the year the fault injection configuration is effective.
+    // 1 represents January, 2 represents February, and so on.
+    Months []int `json:"months,omitempty"`
+}
+
+
+// FaultInjectionState is the status of the fault injection, which may be 'Opened' or 'Closed'.
+type FaultInjectionState string
+
+var (
+	FaultInjectionStatusOpened FaultInjectionState = "Opened"
+	FaultInjectionStatusClosed FaultInjectionState = "Closed"
```

Comment:
`Opend` and `Closed` generally indicate the fuse state. What does this mean for `FaultInjection`? I don't think `FaultInjection` needs to have a snapshot state.

Reply:
A "Fault Injection snapshot" records whether an error code exception has been triggered; it is a snapshot of the occurrence of the exception.